djangorestframework_camel_case/util.py

```diff
@@ -37,10 +37,10 @@ def camelize(data, **options):
                 new_key = re.sub(camelize_re, underscore_to_camel, key)
             else:
                 new_key = key
-            if key not in ignore_fields and new_key not in ignore_fields:
+            if key not in ignore_fields:
                 new_dict[new_key] = camelize(value, **options)
             else:
-                new_dict[new_key] = value
+                new_dict[key] = value
```

Comment:
This change (and this change alone) will fix the issue. It will break the tests however, please see https://github.com/vbabiy/djangorestframework-camel-case/blob/480f67c0266b2c1db6a60418587946169aea14f0/tests.py#L137